### PR TITLE
Warn when possibly using the wrong normalization

### PR DESCRIPTION
**Pull request by ieee8023**
This PR is in response to https://github.com/mlmed/torchxrayvision/issues/89 

When processing an image that may likely not have the correct normalization we can issue a warning. But running min and max on every image/batch is costly so we only do it on the first image/batch.

### CODE REVIEW

File: torchxrayvision/models.py [hunk 1]
```diff
@@ -345,6 +348,26 @@ def fix_resolution(x, resolution: int, model: nn.Module):
...
+    normalization we can issue a warning. But running min and max on 
+    every image/batch is costly so we only do it on the first image/batch.
+    """
+    
+    # Only run this check on the first image so we don't hurt performance.
+    if not "norm_check" in warning_log:
+        x_min = x.min()
+        x_max = x.max()
+        if not torch.logical_and(x_min < -255, x_max > 255):
+            print(f'Warning: Input image does not appear to normalized correctly. The input image has the range [{x_min:.2f},{x_max:.2f}] which doesn\'t seem to be in the [-1024,1024] range. This warning may be wrong though. Only the first image is tested and we are only use a heurstic in an attempt to save a user from using the wrong normalization.')
```
Comment (hbertrand) [inline torchxrayvision/models.py:364]: some typos:
- appear to be
- only using a heuristic

Comment (hbertrand) [inline torchxrayvision/models.py:364]: Should you also raise a warning if there are values lower/greater than -1024 / 1024?

**Comment by hbertrand** [comment]
The correct scaling for #89 seems to be [0 ; 255]. This PR will raise a warning with this range, maybe a mistake?

File: torchxrayvision/models.py (hunk 1 above)
Comment (ieee8023) [reply torchxrayvision/models.py:364]: Oh ya we should also do that. I'll add it.

File: tests/test_models.py [hunk 2]
```diff
@@ -75,3 +75,33 @@ def test_num_classes():
...
         
+def test_normalization_check():
+    
+    models = [xrv.models.DenseNet(weights="densenet121-res224-all"),
+             xrv.models.ResNet(weights="resnet50-res512-all")]
+    
+    incorrect_ranges = [
+        [0, 1],
+        [-1, 1],
+        [0, 1024],
```
[the remark this answers is not in the document]
Comment (ieee8023) [reply tests/test_models.py:86]: I think it would be more likely that someone made a mistake in normalization than an image with no negative values if it was supposed to be [-1024, 1024]. I think it would be better to show a warning and have them verify it is correct than to silently fail.

File: torchxrayvision/models.py (hunk 1 above)
Comment (ieee8023) [reply torchxrayvision/models.py:364]: lol, merci

**Comment by ieee8023** [comment]
> The correct scaling for #89 seems to be [0 ; 255]. This PR will raise a warning with this range, maybe a mistake?

I asked in that issue thread and they scaled to 255 but they should have scaled higher. It reinforces the need for this PR!
